### ISSUE TITLE
Firewall hole punching

### ISSUE THREAD

**Issue by lrettig** [spacemeshos/go-spacemesh#1783]
I'm trying to achieve routability using some sort of proxy as a short-term workaround to the routability issue (see #1544). As discussed with @avive I looked into https://ngrok.com/ which is a very cool tool for TCP but doesn't support UDP.

These tools support TCP _and_ UDP proxying, but it doesn't solve the issue that the node doesn't know what external IP address and port to advertise itself as to other nodes. In order to support this solution, we'd have to add a flag for that.

- https://github.com/txthinking/mr2 (seems to work for me)
- https://github.com/fatedier/frp
- http://zarb.org/~gc/html/udp-in-ssh-tunneling.html
- https://github.com/MarkoPaul0/DatagramTunneler

As another possible workaround, I'm looking into using a VPN instead. That would avoid the issue I just described since the node would be using a different network interface entirely.

**Comment by lrettig** [comment]
Full app sync appears to be working via the VPN I set up (https://github.com/trailofbits/algo), with a range of ports open, but full sync is now also working for me both locally and on a GCE cloud server, both with _no ports open_, after blowing away all data. Odd. I'm planning to meet the Any folks in person tomorrow and will troubleshoot their network connection then. Here's my current thinking:

- Plan A. Things keep magically working without opening ports 🧙‍♂️
- Plan B. Use the VPN
- Plan C. Use an OS X cloud host, RDP or VNC into it
- Plan D. Proxy solution I suggested above - would require code changes
- Plan E. Eliminate need for routability in the first place - eventually we'll want to go this route
